docs/connector-development/connector-builder-ui/authentication.md

```diff
@@ -0,0 +1,185 @@
+# Authentication
+
+Authentication is about providing some form of confidental secret (credentials, password, token, key) to the API provider that identifies the caller (in this case the connector). It allows the provider to check whether the caller is known and has sufficient permission to fetch data. The authentication feature provides a secure way to configure authentication using a variety of methods.
```

Comment:
I think we can get rid of "whether the caller is known". We get the point across just saying:
"It allows the connection to check if it has sufficient permission to fetch data."